bundles/io/org.eclipse.smarthome.io.transport.mqtt/src/main/java/org/eclipse/smarthome/io/transport/mqtt/MqttBrokerConnection.java

```diff
@@ -77,20 +77,22 @@
     };
 
     /// Connection parameters
-    protected Protocol protocol = Protocol.TCP;
+    protected final Protocol protocol;
     protected final String host;
     protected final int port;
     protected final boolean secure;
     protected final String clientId;
     private @Nullable String user;
     private @Nullable String password;
+
     /// Configuration variables
     private int qos = DEFAULT_QOS;
     private boolean retain = false;
     private @Nullable MqttWillAndTestament lastWill;
     private @Nullable AbstractReconnectStrategy reconnectStrategy;
     private SSLContextProvider sslContextProvider = new AcceptAllCertificatesSSLContext();
     private int keepAliveInterval = DEFAULT_KEEPALIVE_INTERVAL;
+    protected @Nullable Map<String, String> websocketHeaders;
```

Comment:
There is now a field and a getter and setter to read / write that map.
But at which place this map is used?
Is an instance of that class now extended to be a "holder" for that map that is used be the consumer?

Reply:
paho just doesn't have the ability to set custom headers yet (https://github.com/eclipse/paho.mqtt.java/pull/514). But I only have small contribution windows and I want this to be in place and not bothered in a few weeks again. We can delay this PR of course.

Reply:
I would prefer to delay it if the Paho version has been bumped as the holding is no useless without the bump.
If you contribution window is closed at the time of the new Paho release that contains your other PR, I assume this could also be done by another one.

Reply:
Yes of course. It should be a one line addition to set the custom headers, next to the paho version bump.